src/get-started/java-binding.md

```diff
@@ -143,71 +143,90 @@ rules – see the corresponding section of our [documentation][transactions].
+the transaction id within that service; a payload containing transaction
+parameters; the public key of the author and a signature that authenticates
+them.
+
+The transaction payload in the message can be serialized
+using an arbitrary algorithm supported by both the service client
+and the service itself.
+
+If a service itself needs to create a transaction on a particular node,
+it can use [`Node#submitTransaction`][node-submit-transaction] method.
+This method will create and sign a transaction message using
+the [service key][node-configuration-validator-keys]
+of _that particular node_ (meaning that the node will be an author
+of the transaction), and submit it into the network.
+Invoking this method on each node unconditionally will produce
+_N_ transactions that have the same payloads, but distinct
+authors’ public keys and signatures, where _N_ is the number of nodes
+in the network.
+
+Ed25519 is a standard cryptographic system for digital signature
+of Exonum messages. It is available through
+[`CryptoFunctions#ed25519`][cryptofunctions-ed25519] method.
 
-!!! note
-    In the current framework version each service must implement a method for
-    sending transactions of this service to the network. It is planned that this
-    mechanism will be standardized and moved to the core.
-    An example of a pseudo-code transaction handler is shown below. For an
-    implementation example, see
-    [`ApiController#submitTransaction`][submittransaction] in the cryptocurrency
-    demo.
-
-!!! note "Example of a Transaction Handler in Pseudo-code"
-    ```none
-    def handle_tx(request):
-      var body = request.getBody()
-      try:
-        var message = txMessageFromBody(body)
-        var tx = service.convertTransaction(txMessage)
-        var hash = tx.hash()
-        node.submitTransaction(tx)
-        return ok(createTxResponse(hash))
-      catch IllegalArgumentException as e:
-        return badRequest("Invalid request: " + e)
-      catch VerificationError as e:
-        return badRequest("Invalid transaction: " + e)
-      catch Error as e:
-        log("Internal error:" + e)
-        return internalServerError()
-    ```
+[node-configuration-validator-keys]: ../architecture/configuration.md#genesis-validator-keys
 
-#### Executable Transactions
+#### Transaction lifecycle
+The lifecycle of a Java service transaction is the same as in any other
+service:
 
-To correctly process a transaction, it must be transformed into an
-*executable transaction* (see
-[`TransactionConverter`][transactionconvererter] or
-`Service#convertToTransaction` method) and transmitted to the framework using
-`Node#submitTransaction` method. The framework verifies it, and if the
-transactions is correct,
-broadcasts it to other nodes of the system. Other nodes, having received the
-transaction message, convert it into an executable transaction, also using the
-service transaction converter.
+1. A service client creates a transaction message, including ids
```

Comment:
All steps have the same number (`1`).

Reply:
The renderer takes care of that. It allows to reorder/remove/add things to the list easily, without rewriting all the subsequent numbers like a monkey :-)

Reply:
It doesn't :astonished: 

@slowli , is it an overly strict linter that forbids using `1.` as in Markdown to specify any item of a numbered list, or is it a renderer itself that cannot put the proper numbers on list items?